src/Datadog.Trace/TraceContext.cs

```diff
@@ -104,9 +105,10 @@ public void CloseSpan(Span span)
 
             lock (_spans)
             {
+                _spans.Add(span);
```

Comment:
@lucaspimentel do the spans need to be ordered in any way? My thinking is no but I want to double check. This would basically add the root span as the last element instead of the first element in the previous implementation

Reply:
That's a good point, gonna check how the agent processes them

Reply:
`GetRoot` looks for the root in the whole trace, so that should be OK: https://github.com/DataDog/datadog-agent/blob/8f141e218ee5bd74d6a7742e73844220c5cda603/pkg/trace/traceutil/trace.go#L27

As a matter of fact, it starts looking by the end, so it looks like we are actually expected to push them in that order

Reply:
> do the spans need to be ordered in any way? 

Not that I know of. 

> it starts looking by the end

Interesting. I wonder how much this affects Trace Agent performance (having tracers add the root span at the start vs end of the array).